### DISCUSSION TITLE
custom play that reads from file

### DISCUSSION THREAD

**Discussion by rock-kobs** [Security-Onion-Solutions/securityonion#7348] [Unsupported Versions]
please how can i create a custom play that reads from a file. i have a ja3 fingerprint file containing malicious hashes creating each custom play will take time, is there another way to do it or am stuck with manual setup.

**Comment by weslambert** [comment]
In this instance, you would need to create a blacklist rule for Elastalert -- at the moment, it might be easier to do this natively, although I'll defer to @defensivedepth to if it is easy enough in Playbook.

https://elastalert.readthedocs.io/en/latest/ruletypes.html#blacklist
https://docs.securityonion.net/en/latest/elastalert.html?highlight=elastalert#configuration

You would also need to ensure the file is accessible by Elastalert.  if you place it in a directory Elastalert already uses, you should be okay. Otherwise, you would need to adjust the Docker bind mounts to allow it.

If you only intend on checking the ja3 fingerprint against a single source of data, it might behoove you to implement the detection using Zeek intel  or something else.

https://docs.securityonion.net/en/latest/zeek.html#intel
